### ISSUE TITLE
CA1717: Incorrect plural detection

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn-analyzers#1846]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

Example: [CA1717](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1717-only-flagsattribute-enums-should-have-plural-names?view=vs-2017)

#### Repro steps

Create the following enumeration:

```csharp
public enum WorkerStatus
{
// Values
}
```

#### Expected behavior

Know dictionary words.

#### Actual behavior

Only check for 's' suffix.
